src/sif/golfcartcost.cc

```diff
@@ -0,0 +1,547 @@
+                              const EdgeLabel& pred) const override;
+
+  /**
+   * Returns the cost to make the transition from the predecessor edge
+   * when using a reverse search (from destination towards the origin).
+   * @param  idx   Directed edge local index
+   * @param  node  Node (intersection) where transition occurs.
+   * @param  pred  the opposing current edge in the reverse tree.
+   * @param  edge  the opposing predecessor in the reverse tree
+   * @param  has_measured_speed Do we have any of the measured speed types set?
+   * @param  internal_turn  Did we make an turn on a short internal edge.
+   * @return  Returns the cost and time (seconds)
+   */
+  virtual Cost TransitionCostReverse(const uint32_t idx,
+                                     const baldr::NodeInfo* node,
+                                     const baldr::DirectedEdge* pred,
+                                     const baldr::DirectedEdge* edge,
+                                     const bool has_measured_speed,
+                                     const InternalTurn internal_turn) const override;
+
+  /**
+   * Get the cost factor for A* heuristics. This factor is multiplied
+   * with the distance to the destination to produce an estimate of the
+   * minimum cost to the destination. The A* heuristic must underestimate the
+   * cost to the destination. So a time based estimate based on speed should
+   * assume the maximum speed is used to the destination such that the time
+   * estimate is less than the least possible time along roads.
+   */
+  virtual float AStarCostFactor() const override {
+    return speedfactor_[top_speed_];
+  }
+
+  /**
+   * Get the current travel type.
+   * @return  Returns the current travel type.
+   */
+  virtual uint8_t travel_type() const override {
+    return static_cast<uint8_t>(VehicleType::kGolfCart);
+  }
+
+  /**
+   * Function to be used in location searching which will
+   * exclude and allow ranking results from the search by looking at each
+   * edges attribution and suitability for use as a location by the travel
+   * mode used by the costing method. It's also used to filter
+   * edges not usable / inaccessible by golf cart.
+   */
+  bool Allowed(const baldr::DirectedEdge* edge,
+               const graph_tile_ptr& tile,
+               uint16_t disallow_mask = kDisallowNone) const override {
+    bool allow_closures = (!filter_closures_ && !(disallow_mask & kDisallowClosure)) ||
+                          !(flow_mask_ & kCurrentFlowMask);
+    return DynamicCost::Allowed(edge, tile, disallow_mask) && !edge->bss_connection() &&
+           (edge->speed_type() == SpeedType::kClassified || tile->edgeinfo(edge).speed_limit() <= max_allowed_speed_limit_) &&
+           (allow_closures || !tile->IsClosed(edge));
+  }
+
+  // Hidden in source file so we don't need it to be protected
+  // We expose it within the source file for testing purposes
+
+  std::vector<float> speedfactor_;
```

Comment:
This should probably be a fixed sized array (open an issue as affects almost all profiles, not just this one).